### ISSUE TITLE
 Visual Studio "GitExt" Menu - "Value does not fall within the expected range."

### ISSUE THREAD

**Issue by laurin1** [gitextensions/gitextensions#5138]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Selecting any item in the menu produces the following error:

![image](https://user-images.githubusercontent.com/75857/42324062-aebbb454-8027-11e8-96b5-9350bb227236.png)

Similar to https://github.com/gitextensions/gitextensions/issues/4708, except the error message.

**Environment you encounter the issue:**

- GitExtensions version: 2.15.02
- GIT version: 2.18.0.windows.1
- OS version: Windows 10 Pro x64 1803
- Visual Studio version info:

Microsoft Visual Studio Community 2015
Version 14.0.25431.01 Update 3
Microsoft .NET Framework
Version 4.7.03056

Installed Version: Community

Visual Basic 2015   00322-20000-00000-AA073
Microsoft Visual Basic 2015

Visual C# 2015   00322-20000-00000-AA073
Microsoft Visual C# 2015

Visual C++ 2015   00322-20000-00000-AA073
Microsoft Visual C++ 2015

Application Insights Tools for Visual Studio Package   7.0.20622.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2015.1   14.1.30801.0
ASP.NET and Web Tools 2015.1

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
For additional information, visit http://www.asp.net/

Automatic Versions   1.11.50
Provides tools for managing assembly versions.

Azure App Service Tools v2.8.1   14.0.11123.0
Azure App Service Tools v2.8.1

Common Azure Tools   1.8
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitExtensions   1.0
Git Extensions is a graphical user interface for Git that allows you to control Git without using the command-line

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

JetBrains ReSharper Ultimate 2018.1    Build 112.0.20180414.70444
JetBrains ReSharper Ultimate package for Microsoft Visual Studio. For more information about ReSharper Ultimate, visit http://www.jetbrains.com/resharper. Copyright © 2018 JetBrains, Inc.

Microsoft .NET Core Tools (Preview 2)   14.1.30801.0
Microsoft .NET Core Tools (Preview 2)

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

NuGet Package Manager   3.5.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

SQL Server Data Tools   14.0.60519.0
Microsoft SQL Server Data Tools

TypeScript   1.8.36.0
TypeScript tools for Visual Studio

**Did this work in previous version of GitExtensions (which)?**

Yes, but it's been a while, but the last time I used VS this worked (about 2 months ago).

I tried uninstalling and re-installing the extension, but still does not work. Also, I just noticed that in the VS Extensions and Updates page, it says 2.15.2:

![image](https://user-images.githubusercontent.com/75857/42324422-c3b1189e-8028-11e8-8b32-647016ec9656.png)

But on the About page for VS, it says 1.0:

![image](https://user-images.githubusercontent.com/75857/42324384-acab1028-8028-11e8-87a7-8b8d4c200e80.png)

**Comment by RussKie** [comment]
Could you please provide at which point you experience this error. E.g. project loaded/not loaded, files opened/not opened etc...

**Comment by laurin1** [comment]
Good point. If I had supplied that info, I would have seen that the issue is more isolated that it originally appeared.

- Project not load - works - opens Git Extensions to Dashboard
- Project loaded, but no document tab - opens Git Extensions to repo
- Project loaded, but with code document tab open and focused - opens Git Extensions to repo
- **Project loaded, but with properties document tab open and focused (regardless of what other tabs are open) - error <--- appears to be the only case that causes this issue**

![image](https://user-images.githubusercontent.com/75857/42380460-a29152b6-80f3-11e8-8495-30f54c739a50.png)

**Comment by drewnoakes** [comment]
I also see this, though with a different error message:

![image](https://user-images.githubusercontent.com/350947/43223597-8669ba76-904b-11e8-86fb-15ec72387a97.png)

The difference may be down to VS version. I'm using VS2017, OP is on 2015.

**Comment by A9G-Data-Droid** [comment]
I can confirm I just installed the extension for the first time today and I get the same error posted by @drewnoakes 7 months ago. He was using VS2017 and the error is the same in VS2019. Closing the property page allows me to use the extension so this thread is helpful in that regard.
